lib/client/client.go

```diff
@@ -288,3 +316,95 @@ func (c *Client) list(metadata unversioned.ResourceMetadata, helper conversionHe
 
 	return nil
 }
+
+// ensureGlobalFelixConfig ensures the specific global felix configuration has
+// been initialized, setting to the default value if not.
+func (c *Client) ensureGlobalFelixConfig(key string, defaultValue interface{}) error {
+	kv := &model.KVPair{
+		Key:   model.GlobalConfigKey{Name: key},
+		Value: fmt.Sprintf("%v", defaultValue),
+	}
+	if _, err := c.backend.Create(kv); err == nil {
+		log.WithField(key, defaultValue).Info("Assigned default global Felix config")
+	} else {
+		if _, ok := err.(errors.ErrorResourceAlreadyExists); !ok {
+			log.WithField(key, defaultValue).WithError(err).Warnf("Failed to set global Felix config")
+			return err
+		}
+		log.WithField(key, defaultValue).Info("Using previously configured global Felix config")
+	}
+	return nil
+}
+
+// ensureGlobalFelixConfig ensures the specific global BGP configuration has
+// been initialized, setting to the default value if not.
+func (c *Client) ensureGlobalBGPConfig(key string, defaultValue interface{}) error {
+	kv := &model.KVPair{
+		Key:   model.GlobalBGPConfigKey{Name: key},
+		Value: fmt.Sprintf("%v", defaultValue),
+	}
+	if _, err := c.backend.Create(kv); err == nil {
+		log.WithField(key, defaultValue).Info("Assigned default global BGP config")
+	} else {
+		if _, ok := err.(errors.ErrorResourceAlreadyExists); !ok {
+			log.WithField(key, defaultValue).WithError(err).Warnf("Failed to set global BGP config")
+			return err
+		}
+		log.WithField(key, defaultValue).Info("Using previously configured global BGP config")
+	}
+	return nil
+}
+
+// ensureGlobalFelixInterfacePrefix ensures the global felix interface prefix
+// contains at least the prefix used by calico/node.
+func (c *Client) ensureGlobalFelixInterfacePrefix() error {
```

Comment:
speaking to Shaun - the default should work, so perhaps I was doing something wrong... I'll retry.  If I do need to include it (temporarily then I'll raise an issue with Felix and remove it once it's fixed).